database/certificate.go

```diff
@@ -272,103 +272,165 @@ func (db *DB) GetCertIDFromTrust(trustID int64) (id int64, err error) {
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(keyusage, &cert.X509v3Extensions.KeyUsage)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(subaltname, &cert.X509v3Extensions.SubjectAlternativeName)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(policies, &cert.X509v3Extensions.PolicyIdentifiers)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(permittednames, &cert.X509v3Extensions.PermittedNames)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(issuer, &cert.Issuer)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(subject, &cert.Subject)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
 	err = json.Unmarshal(key, &cert.Key)
 	if err != nil {
-		return nil, err
+		return cert, err
 	}
 
-	cert.ValidationInfo, cert.Issuer.ID, err = db.GetValidationMapForCert(certID)
-
+	cert.ValidationInfo, cert.Issuer.ID, err = db.GetValidationMapForCert(cert.ID)
 	return cert, err
+}
 
+// GetCertByID fetches a certain certificate from the database.
+// It returns a pointer to a Certificate struct and any errors that occur.
+func (db *DB) GetCertByID(certID int64) (*certificate.Certificate, error) {
+	row := db.QueryRow(`SELECT
+                        id,
+			serial_number,
+			sha1_fingerprint,
+			sha256_fingerprint,
+			sha256_subject_spki,
+			pkp_sha256,
+			issuer,
+			subject,
+			version,
+			is_ca,
+			not_valid_before,
+			not_valid_after,
+			key,
+			first_seen,
+			last_seen,
+			x509_basicConstraints,
+			x509_crlDistributionPoints,
+			x509_extendedKeyUsage,
+			x509_authorityKeyIdentifier,
+			x509_subjectKeyIdentifier,
+			x509_keyUsage,
+			x509_subjectAltName,
+			x509_certificatePolicies,
+			is_name_constrained,
+			permitted_names,
+			signature_algo,
+			raw_cert
+		FROM certificates WHERE id=$1`, certID)
+	cert, err := db.scanCert(row)
+	return &cert, err
+
+}
+
+var ErrInvalidCertStore = fmt.Errorf("Invalid certificate store provided")
+
+func (db *DB) GetAllCertsInStore(store string) (out []certificate.Certificate, err error) {
+	switch store {
+	case
+		"mozilla",
+		"android",
+		"apple",
+		"microsoft",
+		"ubuntu":
+		query := fmt.Sprintf(`SELECT
+                        id,
+			serial_number,
+			sha1_fingerprint,
+			sha256_fingerprint,
+			sha256_subject_spki,
+			pkp_sha256,
+			issuer,
+			subject,
+			version,
+			is_ca,
+			not_valid_before,
+			not_valid_after,
+			key,
+			first_seen,
+			last_seen,
+			x509_basicConstraints,
+			x509_crlDistributionPoints,
+			x509_extendedKeyUsage,
+			x509_authorityKeyIdentifier,
+			x509_subjectKeyIdentifier,
+			x509_keyUsage,
+			x509_subjectAltName,
+			x509_certificatePolicies,
+			is_name_constrained,
+			permitted_names,
+			signature_algo,
+			raw_cert
+                    FROM certificates WHERE in_%s_root_store=true`, store)
```

Comment:
As an attacker, how would you exploit this? Keep in mind the switch validates the store variable.